File: api/src/endpoint/migration/get_migration.rs
use crate::{endpoint::Endpoint, resource::migration::MigrationResource};

pub struct GetMigration;

impl Endpoint for GetMigration {
    const PATH: &'static str = "/migration/{number}";
    const METHOD: http::Method = http::Method::GET;

    type Request = ();
    type Response = GetMigrationResponse;
}

pub type GetMigrationResponse = MigrationResource;

File: gitdot-api-ts/src/endpoint/migration/get-migration.ts
import { z } from "zod";

import { MigrationResource } from "../../resource";

export const GetMigrationRequest = z.object({});
export type GetMigrationRequest = z.infer<typeof GetMigrationRequest>;

export const GetMigration = {
  path: "/migration/{number}",
  method: "GET",
  request: GetMigrationRequest,
  response: MigrationResource,
} as const;
export type GetMigration = typeof GetMigration;
